### PR TITLE
Disable pagechange event if page is not changing.

### LINKED ISSUE #7288: pagechange event fires two times when user changes page
Link to PDF file (or attach file here):
Base file at https://mozilla.github.io/pdf.js/web/viewer.html

Configuration:
- Web browser and its version: Chrome 50
- Operating system and its version: Mac OSX 10.10.5
- PDF.js version: 1.5.230
- Is an extension: No

Steps to reproduce the problem:
1. Open devtools and add a breakpoint in viewer.js (currently line 2741):
```
eventBus.on('pagechange', function (e) {
  var event = document.createEvent('UIEvents'); // <-- Add breakpoint here
  event.initUIEvent('pagechange', true, true, window, 0);
  event.updateInProgress = e.updateInProgress;
  event.pageNumber = e.pageNumber;
  event.previousPageNumber = e.previousPageNumber;
  e.source.container.dispatchEvent(event);
});
```

2. Change page number using toolbar, say to page 2 of 14.
3. See that breakpoint is triggered twice

What is the expected behavior? (add screenshot)
'pagechange' event should be fired only once

What went wrong? (add screenshot)
'pagechange' event is fired twice, once from the setting of currentPageNumber, and another when the screen updates.
1st time: http://screencast.com/t/kdjojgK0cC
2nd time: http://screencast.com/t/9ssbszRNQ

Link to a viewer (if hosted on a site other than mozilla.github.io/pdf.js or as Firefox/Chrome extension):
https://mozilla.github.io/pdf.js/web/viewer.html

### PR DESCRIPTION
**Pull request by yurydelendik**
Fixes #7288

### CODE REVIEW

**Comment by yurydelendik** [comment]
Let's see if it breaks our viewer

/botio-windows preview

**Comment by yurydelendik** [comment]
- [ ] check with mozcentral tests

**Comment by Snuffleupagus** [comment]
This seems to cause some glitches related to Presentation Mode, e.g
1. Open the preview.
2. Go to the first page.
3. Enter Presentation Mode.
4. Exit Presentation Mode.

Note how the *top* of the first page is now scrolled down a bit, when compared with the same steps in http://mozilla.github.io/pdf.js/web/viewer.html.

**Comment by Snuffleupagus** [comment]
The title of the PR is "Disable pagechange event if page is not changing.", but the code actually goes a bit further than that. If the intent of the patch is simply to not fire the event unnecessarily, then we probably still need https://github.com/mozilla/pdf.js/blob/master/web/pdf_viewer.js#L192-L196 even in the `this._currentPageNumber === val` case.
Please note that I've not tested it, but I'm guessing that that code would fix the the Presentation Mode glitch that I described above, and might also prevent other issues where existing code relies on the current behaviour.

**Comment by yurydelendik** [comment]
> then we probably still need https://github.com/mozilla/pdf.js/blob/master/web/pdf_viewer.js#L192-L196 even in the this._currentPageNumber === val case.

That what I was afraid of. We need to remove this logic from currentPageNumber somehow.

**Comment by yurydelendik** [comment]
I was not able to reproduce to STR above. However I agree that not calling this.scrollPageIntoView might break things. I refactored it further to remove updateInProgress and extracted _resetCurrentPageView -- now the logic must be preserved.

/botio-windows preview

File: web/pdf_viewer.js [hunk 1]
```diff
@@ -673,13 +676,11 @@ var PDFViewer = (function pdfViewer() {
       }
 
       if (!this.isInPresentationMode) {
-        this.currentPageNumber = currentId;
+        this._setCurrentPageNumber(currentId, true);
```
Comment (yurydelendik) [inline web/pdf_viewer.js]: this might cause the confusion -- we need to rethink this logic in the future

File: web/pdf_viewer.js [hunk 2]
```diff
@@ -165,12 +164,21 @@ var PDFViewer = (function pdfViewer() {
         this._currentPageNumber = val;
         return;
       }
+      this._setCurrentPageNumber(val, false);
+      // The intent can be to just reset a scroll position and/or scale.
+      this._resetCurrentPageView();
+    },
 
+    _setCurrentPageNumber:
+        function pdfViewer_setCurrentPageNumber(val, updateInProgress) {
```
Comment (Snuffleupagus) [inline web/pdf_viewer.js]: I'm wondering if we actually still need the `updateInProgress` parameter here, and in the events?
Previously we called `scrollPageIntoView` from the event handler in `app.js` (or `viewer.js` as it were), which seemed bad for e.g. the viewer components. In PR #6140, I removed some dead code, and also refactored this. With the changes in this patch, it seems to me at least, that `updateInProgress` is now pointless.
(Obviously there might be someone relying on it, but given that we've not really made any promises about the stability of the viewer components, I think we can remove it.)

Comment (yurydelendik) [inline web/pdf_viewer.js]: Agree.

**Comment by Snuffleupagus** [comment]
I can confirm that issue I reported in https://github.com/mozilla/pdf.js/pull/7289#issuecomment-216791562 is now fixed.

Looks good, thank you for the patch!

**Pull request merged**